src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Callable
+
+from databricks.sdk.service.workspace import ObjectType, ObjectInfo, ExportFormat
+from databricks.sdk import WorkspaceClient
+
+
+class Dependency:
+
+    @staticmethod
+    def from_object_info(object_info: ObjectInfo):
+        assert object_info.path is not None
+        return Dependency(object_info.object_type, object_info.path)
+
+    def __init__(self, object_type: ObjectType | None, path: str):
+        self._type = object_type
+        self._path = path
+
+    @property
+    def type(self) -> ObjectType | None:
+        return self._type
+
+    @property
+    def path(self) -> str:
+        return self._path
+
+    def __hash__(self):
+        return hash(self.path)
+
+    def __eq__(self, other):
+        return isinstance(other, Dependency) and self.path == other.path
+
+
+class SourceContainer(abc.ABC):
+
+    @property
+    @abc.abstractmethod
+    def object_type(self) -> ObjectType:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def build_dependency_graph(self, graph) -> None:
+        raise NotImplementedError()
+
+
+class DependencyLoader:
```

Comment:
dependency loader has to be an interface. There has to be `LocalDependencyLoader` and `WorkspaceDependencyLoader`

Reply:
well the intent is to manage both workspace and local dependencies from here...